ownCloud/App Controllers/AppRootViewController.swift

```diff
@@ -116,7 +118,46 @@ open class AppRootViewController: EmbeddingViewController, BrowserNavigationView
 		contentBrowserController.sidebarViewController = leftNavigationController
 
 		// Make browser navigation view controller the content
-		contentViewController = contentBrowserController
+		noBookmarkCondition = DataSourceCondition(.empty, with: OCBookmarkManager.shared.bookmarksDatasource, initial: true, action: { [weak self] condition in
+			if condition.fulfilled == true {
+				// No account available
+                let messageView = ComposedMessageView.infoBox(additionalElements: [
+                    .image(AccountSettingsProvider.shared.logo, size: CGSize(width: 128, height: 128)),
+                    .title(String(format: "Welcome to %@".localized, VendorServices.shared.appName), alignment: .centered, insets: NSDirectionalEdgeInsets(top: 25, leading: 0, bottom: 25, trailing: 0))
+                ])
+                messageView.cssSelector = .info
+                messageView.backgroundView?.cssSelector = .info
+                messageView.elementInsets = NSDirectionalEdgeInsets(top: 25, leading: 50, bottom: 50, trailing: 50)
+                
+                var addAccountTitle = "Add account".localized
+                if !VendorServices.shared.canAddAccount {
+                    addAccountTitle = "Login".localized
+                }
+                
+                messageView.elements?.append(.button(addAccountTitle, action: UIAction(handler: { [weak self] action in
```

Comment:
The buttons could also be included directly in the array passed to `.infoBox(additionalElements:)`.